compiler/noirc_evaluator/src/acir/arrays.rs

```diff
@@ -508,11 +508,7 @@ impl Context<'_> {
         // and may contain data for which we want to restrict access. The true slice length is tracked in a
         // a separate SSA value and restrictions on slice indices should be generated elsewhere in the SSA.
         let array_typ = dfg.type_of_value(array);
-        let array_len = if !array_typ.contains_slice_element() {
-            array_typ.flattened_size() as usize
-        } else {
-            self.flattened_slice_size(array, dfg)
-        };
```

Comment:
We have this if-statement popping up in a few places so it got turned into a helper function.